guides/developer/getting-set-up.md

```diff
@@ -0,0 +1,108 @@
+---
+layout: single
+title: "Getting Set Up for Spinnaker Development"
+sidebar:
+  nav: guides
+---
+
+{% include toc %}
+
+This page describes the steps a Developer should take to fetch Spinnaker's codebase
+and get set up to work on it.
+
+Follow the [contributing guidelines](/community/contributing/submitting/)
+if you plan to submit your work as a patch to the open source project.
+
+# System Requirements
+
+This guide assumes you have access to a machine with a minimum specification of:
+
+- 18 GB of RAM
+- A 4 core CPU
+- Ubuntu 14.04, 16.04 or 18.04
+
+> This guide has been tested against a machine with these specifications but it's feasible
+> to develop Spinnaker on different flavors of Linux and with fewer resources, depending on
+> what you're working on.
+
+# Fetching Spinnaker's codebase
+
+The following steps will install Spinnaker's management tool, Halyard, fetch Spinnaker's
+codebase, and perform just enough configuration to get Spinnaker up and running.
+
+Each of these steps will take between 5 and 30 minutes to finish and may require you to
+do multiple things, such as installing CLI tools or configuring service accounts. A good
+way to approach this process is to open a step in its own browser tab and then work through
+it to completion, closing it (and any others you've opened in support of it) once it's done.
+
+1. [Install Halyard](/setup/install/halyard/#install-on-debianubuntu-and-macos)
+1. [Set up a storage service](/setup/install/storage/)
+1. [Set up your cloud provider of choice](/setup/install/providers)
+1. [Configure a LocalGit deployment](/setup/install/environment/#local-git)
+1. Run `hal deploy apply`
+
+## What does this do?
+
+Halyard creates a directory at `~/dev/spinnaker` with the following contents:
+- a subdirectory for each Spinnaker service containing that service's source code
+- a `scripts` directory
+- a `logs` directory
+- a set of `.pid` files, one for each service that is running
+
+The `scripts` directory contains scripts to start and stop each individual service. For
+example calling `~/dev/spinnaker/scripts/deck-stop.sh` will kill the Deck process and
+delete the `deck.pid` file. Running `~/dev/spinnaker/scripts/deck-start.sh` will restart Deck
+and recreate its `deck.pid` file.
+
+The final step, `hal deploy apply`, checks out the git repos for each service and launches
```

Comment:
Worth mentioning that it also populates ~/.spinnaker with each service's config file?

Reply:
Good idea, I've added a paragraph around this and `~/.hal`